packages/utils/tokenAllowance/useHandleTokenApprove.ts

```diff
@@ -1,13 +1,16 @@
-import { useIFrameContext } from "#/context/iframe";
-import { TransactionFactory } from "#/utils/transactionFactory/factory";
-import { TRANSACTION_TYPES } from "#/utils/transactionFactory/types";
-import { MAX_UINT256 } from "@balancer/sdk";
+import { TRANSACTION_TYPES, TransactionFactory } from "../transactionFactory";
+import type { Signer } from "ethers";
 import { useCallback } from "react";
-import { Address } from "viem";
-
-export function useHandleTokenApprove() {
-  const { signer, cowShedProxy } = useIFrameContext();
+import type { Address } from "viem";
+import { MAX_UINT256 } from "@balancer/sdk";
 
+export function useHandleTokenApprove({
+  signer,
+  cowShedProxy,
+}: {
+  signer: Signer | undefined;
+  cowShedProxy: Address | undefined;
```

Comment:
Approving to the cowShedProxy is already an implementation of the useHandleTokenApprove in my opinion hook. Maybe we could refactor to `spender`.